libr/core/cmd.c

```diff
@@ -1266,6 +1282,239 @@ static int cmd_stdin(void *data, const char *input) {
 	return r_core_run_script (core, "-");
 }
 
+static void load_table(RCore *core, RTable *t, const char *file) {
+	char *data = NULL;
+
+	if (*file == '$') {
+		const char *cdata = r_cmd_alias_get (core->rcmd, file, 1);
+		if (cdata) {
+			data = strdup (cdata + 1);
+		}
+	} else {
+		data = r_file_slurp (file, NULL);
+	}
+	if (!data) {
+		eprintf ("Error: Cannot slurp '%s'\n", file);
+		return;
+	}
+	RListIter *iter;
+	char *line;
+	RList *lines = r_str_split_list (data, "\n", 0);
+	bool expect_header = false;
+	bool expect_rows = false;
+	const char *separator = "|";
+	int ncols = 0;
+	r_list_foreach (lines, iter, line) {
+		if (!expect_rows) {
+			if (r_str_startswith (line, ".--")) {
+				expect_header = true;
+				separator = "|";
+				continue;
+			}
+			if (r_str_startswith (line, "┌")) {
+				expect_header = true;
+				separator = "│";
+				continue;
+			}
+			if (r_str_startswith (line, ")-")) {
+				expect_rows = true;
+				separator = "|";
+				continue;
+			}
+			if (r_str_startswith (line, "│─")) {
+				expect_rows = true;
+				separator = "│";
+				continue;
+			}
+		}
+
+		RTableColumnType *typeString = r_table_type ("string");
+		RTableColumnType *typeNumber = r_table_type ("number");
+		if (expect_header) {
+			char *arg;
+			RList *args = r_str_split_list (line + strlen (separator), separator, 0);
+			RListIter *iter2;
+			ncols = 0;
+			if (r_list_length (t->cols) > 0) {
+				eprintf ("Warning: Not re-adding headers. Use ,- to reset the table.\n");
+				continue;
+			}
+			r_list_foreach (args, iter2, arg) {
+				char *s = strchr (arg, ' ');
+				char *ss = r_str_trim_dup (s? s + 1: arg);
+				if (!*ss) {
+					free (ss);
+					continue;
+				}
+				r_table_add_column (t, typeString, ss, 0);
+				ncols ++;
+			}
+			expect_header = false;
+		} else if (expect_rows) {
+			char *arg;
+			RList *args = r_str_split_list (line + strlen (separator), separator, 0);
+			RList *items = r_list_newf (free);
+			RListIter *iter2;
+			if (r_list_length (args) < ncols) {
+				// dowarn?
+				continue;
+			}
+			r_list_foreach (args, iter2, arg) {
+				char *ss = r_str_trim_dup (arg);
+				if (!*ss) {
+					free (ss);
+					continue;
+				}
+				if (isdigit (*ss)) {
+					int col = r_list_length (items);
+					RTableColumn *c = r_list_get_n (t->cols, col);
+					if (c) {
+						c->type = typeNumber;
+					}
+				}
+				r_list_append (items, ss);
+			}
+			RTableRow *row = r_table_row_new (items);
+			r_list_append (t->rows, row);
+		}
+	}
+	r_list_free (lines);
+	free (data);
+}
+
+static int cmd_table(void *data, const char *input) {
+	RCore *core = (RCore*)data;
+
+	if (!core->table) {
+		core->table = r_table_new ();
+	}
+
+	switch (*input) {
```

Comment:
could you split each case like how it is done in https://github.com/radareorg/radare2/blob/master/libr/core/cmd_write.c#L1981 ?

Reply:
Yes, but, if the reason is readability it will be better to move the whole thing into a separate .c file. if its for api accessibility thats a different topic, all of them or only the ones that hold local vars between braces?

Reply:
done

Reply:
so do you agree on adding cmd_table.c and adding all those static functions in there? @ret2libc

Reply:
It's also ok as-is, don't worry.